packages/components/src/confirm-dialog/component.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import { forwardRef, useEffect, useState } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import type { Ref, SyntheticEvent, KeyboardEvent } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Modal from '../modal';
+import type { OwnProps } from './types';
+import {
+	useContextSystem,
+	contextConnect,
+	WordPressComponentProps,
+} from '../ui/context';
+import { Flex } from '../flex';
+import Button from '../button';
+import * as styles from './styles';
+import { useCx } from '../utils/hooks/use-cx';
+
+function ConfirmDialog(
+	props: WordPressComponentProps< OwnProps, 'div', false >,
+	forwardedRef: Ref< any >
+) {
+	const {
+		isOpen,
+		message,
+		onConfirm,
+		onCancel = () => {},
+		...otherProps
+	} = useContextSystem( props, 'ConfirmDialog' );
+
+	const cx = useCx();
+	const wrapperClassName = cx( styles.wrapper );
+
+	const [ _isOpen, setIsOpen ] = useState< Boolean >();
+	const [ _selfClose, setSelfClose ] = useState< Boolean >();
```

Comment:
- Let's use `boolean` (lowercase) instead.
- Let's also avoid naming variables with a trailing underscore. To solve the name conflict, you could rename the `isOpen` prop to `isOpenProp`

```suggestion
	const {
		isOpen: isOpenProp,
		message,
		onConfirm,
		onCancel = () => {},
		...otherProps
	} = useContextSystem( props, 'ConfirmDialog' );

	const cx = useCx();
	const wrapperClassName = cx( styles.wrapper );

	const [ isOpen, setIsOpen ] = useState< boolean >();
	const [ selfClose, setSelfClose ] = useState< boolean >();
```

Reply:
> Let's use boolean (lowercase) instead.
Oh, thanks for pointing that out! I shouldn't have used the `Boolean` object there, indeed.

Reply:
> Let's also avoid naming variables with a trailing underscore. To solve the name conflict, you could rename the isOpen prop to isOpenProp

Sounds good!